pkg/network/encoding/encoding.go

```diff
@@ -50,6 +50,37 @@ func GetMarshaler(accept string) Marshaler {
 	return jSerializer
 }
 
+// ConnectionsModeler contains all the necessary structs for modeling a connection.
+type ConnectionsModeler struct {
+	httpEncoder  *httpEncoder
+	http2Encoder *http2Encoder
+	kafkaEncoder *kafkaEncoder
+	dnsFormatter *dnsFormatter
+	ipc          ipCache
+}
+
+// NewConnectionsModeler initializes the connection modeler with encoders, dns formatter for
+// the existing connections. The ConnectionsModeler holds the traffic encoders grouped by USM logic.
+// It also includes formatted connection telemetry related to all batches, not specific batches.
+// Furthermore, it stores the current agent configuration which applies to all instances related to the entire set of connections,
+// rather than just individual batches.
+func NewConnectionsModeler(conns *network.Connections) *ConnectionsModeler {
+	return &ConnectionsModeler{
+		httpEncoder:  newHTTPEncoder(conns.HTTP),
+		http2Encoder: newHTTP2Encoder(conns.HTTP2),
+		kafkaEncoder: newKafkaEncoder(conns.Kafka),
+		ipc:          make(ipCache, len(conns.Conns)/2),
+		dnsFormatter: newDNSFormatter(conns, make(ipCache, len(conns.Conns)/2)),
+	}
+}
+
+// Close cleans all encoders resources.
```

Comment:
what about dns?

Reply:
Actually I do not see any related reset/close functions for the `dnsFormatter` as the decoders have